File: backend/app/api/routes/baseline_parity.py
from fastapi import APIRouter, Depends, HTTPException
from app.api.baseline_parity_contracts import BaselineParityCaptureRequest, BaselineParityResponse
from app.services.baseline_parity_application_service import BaselineParityApplicationError, BaselineParityApplicationService

router = APIRouter(prefix="/runs", tags=["baseline-parity"])


def get_service() -> BaselineParityApplicationService:
    return BaselineParityApplicationService()


def _raise(error: BaselineParityApplicationError):
    raise HTTPException(status_code=error.status_code, detail={"error_code": error.code, "message": error.message})


@router.post("/{run_id}/baseline/parity", response_model=BaselineParityResponse)
def capture(run_id: str, request: BaselineParityCaptureRequest, service: BaselineParityApplicationService = Depends(get_service)):
    try:
        return service.capture(run_id, request)
    except BaselineParityApplicationError as error:
        _raise(error)


def _get(run_id: str, section: str, service: BaselineParityApplicationService):
    result = service.get(run_id, section)
    if result is None:
        raise HTTPException(status_code=404, detail={"error_code": "BASELINE_PARITY_NOT_FOUND", "message": "Baseline parity evidence was not found."})
    return result


@router.get("/{run_id}/baseline/failures", response_model=BaselineParityResponse)
def failures(run_id: str, service: BaselineParityApplicationService = Depends(get_service)):
    return _get(run_id, "failures", service)


@router.get("/{run_id}/baseline/routes", response_model=BaselineParityResponse)
def routes(run_id: str, service: BaselineParityApplicationService = Depends(get_service)):
    return _get(run_id, "routes", service)


@router.get("/{run_id}/baseline/backend-integration", response_model=BaselineParityResponse)
def backend_integration(run_id: str, service: BaselineParityApplicationService = Depends(get_service)):
    return _get(run_id, "backend-integration", service)


@router.get("/{run_id}/baseline/anchors", response_model=BaselineParityResponse)
def anchors(run_id: str, service: BaselineParityApplicationService = Depends(get_service)):
    return _get(run_id, "anchors", service)

File: frontend/src/api/baselineParity.ts
import { apiClient, type createApiClient } from "./client";
import type { BaselineParityCaptureRequest, BaselineParityResponse } from "@/types/baselineParity";

type ApiClient = ReturnType<typeof createApiClient>;
function runPath(runId: string) { return encodeURIComponent(runId); }

export function captureBaselineParity(runId: string, request: BaselineParityCaptureRequest, client: ApiClient = apiClient): Promise<BaselineParityResponse> {
  return client.post<BaselineParityResponse>(`/api/v1/runs/${runPath(runId)}/baseline/parity`, request);
}

export function getBaselineParitySection(runId: string, section: "failures" | "routes" | "backend-integration" | "anchors", client: ApiClient = apiClient): Promise<BaselineParityResponse> {
  return client.get<BaselineParityResponse>(`/api/v1/runs/${runPath(runId)}/baseline/${section}`);
}
